modules/cudacodec/include/opencv2/cudacodec.hpp

```diff
@@ -152,14 +152,14 @@ class CV_EXPORTS EncoderCallBack
     @param frameNumber
     @param picType Specify frame type (I-Frame, P-Frame or B-Frame).
      */
-    virtual void onBeginFrame(int frameNumber, PicType picType) = 0;
+    CV_WRAP virtual void onBeginFrame(int frameNumber, EncoderCallBack::PicType picType) = 0;
 
```

Comment:
The wrapper converts the arguments into statements, thus enum arguments results in maybe-uninitialized warnings as in [this build](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/814/steps/compile%20release/logs/warnings%20%284%29).

I wonder which resolution is better...
- Arguments should be converted to integers
- The generator should  do better conversion
- Maybe-uninitialized warnings of the generator code should be silenced

Reply:
What is about forced initialization of these enum values in generator code?
```.diff
-EncoderCallBack::PicType picType;
+EncoderCallBack::PicType picType = (EncoderCallBack::PicType)0;
```

Reply:
The question is how to tell which argument is enum and which is not.

Reply:
I totally agree, and hope we can do that in the near future.
However, for the time being, I gave up on binding these functions to avoid their warnings.

Reply:
@alalek Today, I have supported enum-type detection. I hope you would consider merging this PR before it gets any bigger...